src/include/rbd/librbd.h

```diff
@@ -948,6 +948,9 @@ CEPH_RBD_API int rbd_snap_set_limit(rbd_image_t image, uint64_t limit);
 CEPH_RBD_API int rbd_snap_get_timestamp(rbd_image_t image, uint64_t snap_id, struct timespec *timestamp);
 
 CEPH_RBD_API int rbd_snap_set(rbd_image_t image, const char *snapname);
+CEPH_RBD_API int rbd_snap_set_by_namespace_type(rbd_image_t image,
+                                                const char *snap_name,
+                                                rbd_snap_namespace_type_t snap_ns_type);
```

Comment:
As discussed offline, adding `snap_set_by_namespace_type` API is probably not needed.